code/game/objects/random/random.dm

```diff
@@ -397,18 +397,18 @@
 	)
 	has_postspawn = TRUE
 
-/obj/random/voidsuit/New(var/_damaged = 0)
+/obj/random/voidsuit/New(loc, _damaged = 0)
 	damaged = _damaged
-	..()
+	..(loc)
 
 /obj/random/voidsuit/post_spawn(obj/item/clothing/suit/space/void/suit)
-	var/helmet = suitmap[suit]
+	var/helmet = suitmap[suit.type]
 	if (helmet)
 		new helmet(loc)
 	else
 		log_debug("random_obj (voidsuit): Type [suit.type] was unable to spawn a matching helmet!")
 	new /obj/item/clothing/shoes/magboots(loc)
-	if (damaged & prob(60))
+	if (damaged && prob(60))
```

Comment:
Just so you know. This was the only error IIRC. The loc thing should have been fine, there's no need to have it explicitly written. Calling `..()` as is will pass all arguments to it.